### PR TITLE
9 as a user i want to set all supported fields on collection create

### LINKED ISSUE #9: [BLOCKED] As a user, I want to set all supported fields on collection create
Currently missing:
- Expiration Time
- Auto Renew Account
- Auto Renew Period
- Memo

### PR DESCRIPTION
**Pull request by BartoszSolkaArianelabs**
**Description**:
Adds support to missing fields in createCollection

⚠️ not yet fully functional as the autoRenewAccount is bugged in Hedera SDK. This causes one end2end test to fail. Let's discuss if we want to merge it and disable the test for now or do you have any other ideas

**Related issue(s)**:

Fixes #9

### CODE REVIEW

File: src/utils/validate-props.ts [hunk 1]
```diff
@@ -112,6 +113,17 @@ const validateAccountAndPrivateKey = (props: validateCreateCollectionProps) => {
...
+  if (
+    Object.prototype.hasOwnProperty.call(props, 'autoRenewAccount') ||
+    Object.prototype.hasOwnProperty.call(props, 'autoRenewAccountPrivateKey')
+  ) {
+    if ((props.autoRenewAccount && !props.autoRenewAccountPrivateKey) || (!props.autoRenewAccount && props.autoRenewAccountPrivateKey)) {
+      throw new Error(dictionary.createCollection.autoRenewAccountPrivateKeySignRequired);
+    }
+  }
+};
+
```
Comment (michalrozekariane) [inline src/utils/validate-props.ts:126]: I think it's more clear to destructure props like this:
```
const validateAutoRenewAccount = ({
  autoRenewAccount,
  autoRenewAccountPrivateKey,
}: ValidateCreateCollectionProps) => {
```

File: src/test/e2e/create-collection-e2e.test.ts [hunk 2]
```diff
@@ -68,5 +68,83 @@ describe('createCollectionFunction e2e', () => {
...
-  });
+  }, LONG_E2E_TIMEOUT);
+
+  it('creates a collection with expiration time', async () => {
+    const tokenId = await nftSDK.createCollection({
+      collectionName: 'test_name_expiration_time',
+      collectionSymbol: 'TNET',
+      treasuryAccountPrivateKey: secondPrivateKey,
+      treasuryAccount: secondAccountId,
+      expirationTime: new Date(Date.now() + 1 * 24 * 60 * 60 * 1000),// now + 1 day
```
Comment (StanislawMazowieckiAriane) [inline src/test/e2e/create-collection-e2e.test.ts:79]: Next time use [datasnfs](https://date-fns.org/), or if you have time now do it.
Operations directly on a date are unstable and irritable.
Hard to read, hard to understand, maintain, etc.

File: src/test/e2e/create-collection-e2e.test.ts [hunk 3]
```diff
@@ -68,5 +68,83 @@ describe('createCollectionFunction e2e', () => {
...
+    });
+
+    const tokenInfo = await getTokenInfo(tokenId, nftSDK.client);
+    const nowPlusWeek = new Date(Date.now() + 7 * 24 * 60 * 60 * 1000);
```
Comment (StanislawMazowieckiAriane) [inline src/test/e2e/create-collection-e2e.test.ts:83]: Same as above

**Review by se7enarianelabs — APPROVED** [review]

File: src/test/e2e/create-collection-e2e.test.ts (hunk 2 above)
Comment (BartoszSolkaArianelabs) [reply src/test/e2e/create-collection-e2e.test.ts]: done

File: src/test/e2e/create-collection-e2e.test.ts (hunk 3 above)
Comment (BartoszSolkaArianelabs) [reply src/test/e2e/create-collection-e2e.test.ts]: done

File: src/test/e2e/create-collection-e2e.test.ts [hunk 4]
```diff
@@ -68,5 +68,83 @@ describe('createCollectionFunction e2e', () => {
...
+    expect(tokenInfo.expirationTime).toBeDefined();
+    expect(tokenInfo.expirationTime!.toDate().getTime() < nowPlusWeek.getTime()).toBeTruthy();
+  }, LONG_E2E_TIMEOUT);
+
+  it('creates a collection with auto renew account and period', async () => {
+    const expectedAutoRenewPeriod = 2592000;// 30 days
```
[the remark this answers is not in the document]
Comment (BartoszSolkaArianelabs) [reply src/test/e2e/create-collection-e2e.test.ts]: done

**Review by StanislawMazowieckiAriane — APPROVED** [review]

**Pull request merged**